app/services/dossier_projection_service.rb

```diff
@@ -0,0 +1,82 @@
+class DossierProjectionService
+  TABLE = 'table'
+  COLUMN = 'column'
+
+  # Returns [Openstruct(dossier, columns)] ordered by dossiers_ids
+  # and the columns orderd by fields.
+  #
+  # It tries to be fast by using pluck (or at least select) method
+  # to avoid deserializing whole AR object.
+  #
+  # It stores its intermediary queries results in an hash in the corresponding field.
+  # ex: field_email[:id_value_h] = { dossier_id_1: email_1, dossier_id_3: email_3 }
+  #
+  # Those hashes are needed as the order of the intermediary query results are unknown
+  # and some value can be missing du to different revision.
```

Comment:
```suggestion
  # Those hashes are needed because:
  # - the order of the intermediary query results are unknown,
  # - some values can be missing (if a revision added or removed them).
```